### PR TITLE
feat: add autocomplete support

### PR DESCRIPTION
**Pull request by ExampleWasTaken**
## Description
This PR adds support for the autocomplete feature of slash commands.

The implementation closely follows the already existing infrastructure by adding a third (and optional) argument for an autocomplete-callback function to the `slashCommand()` function.

## Test Results
This PR has no effect on existing code. It only lays the groundwork for future commands that may require autocompletion.

## Additional Information
This PR is required for #51.

## Discord Username
examplewastaken

### CODE REVIEW

**Review by benw202 — COMMENTED** [review]
Please add `src/lib/autocomplete.ts` to the index.ts file in the lib dir.

You will then need to change your import in slashCommand.ts

**Comment by ExampleWasTaken** [comment]
Yep good point. Will do.

File: src/lib/autocomplete.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { AutocompleteInteraction, Awaitable, Client } from 'discord.js';
+import { LogMethods } from './events';
```
Comment (benw202) [inline src/lib/autocomplete.ts]:
```suggestion
import { LogMethods } from './index';
```

File: src/lib/slashCommand.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import type { Awaitable, Client, ChatInputCommandInteraction, SlashCommandBuilder, SlashCommandSubcommandsOnlyBuilder, RESTPostAPIApplicationCommandsJSONBody } from 'discord.js';
 import { LogMethods } from './index';
```
Comment (benw202) [inline src/lib/slashCommand.ts]:
```suggestion
import { LogMethods, AutocompleteCallback } from './index';
```

File: src/lib/slashCommand.ts [hunk 3]
```diff
@@ -1,5 +1,6 @@
...
+import { AutocompleteCallback } from './index';
```
Comment (benw202) [inline src/lib/slashCommand.ts]:
```suggestion
```

**Review by benw202 — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -225,16 +237,31 @@ const data = slashCommandStructure({
...
 });
 
+// Optional - This is only required if you plan on using autocomplete for your command.
+const autocompleteCallback: AutocompleteCallback = ({ interaction }) => {
+    // Autocomplete logic goes here
+    const choices = [
+        {
+            name: 'Display Name',
+            value: 'Value that is actually sent to your bot.'
+        }
```
Comment (benw202) [inline .github/CONTRIBUTING.md]:
```suggestion
        },
```

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -225,16 +237,31 @@ const data = slashCommandStructure({
...
+    ];
+
+    // If you can't autocomplete anything respond with an empty array.
+    return interaction.respond(choices);
+}
```
Comment (benw202) [inline .github/CONTRIBUTING.md]:
```suggestion
};
```

**Review by benw202 — APPROVED** [review]

**Review by pdellaert — APPROVED** [review]

**Pull request merged**